internal/provider/resource_mackerel_channel.go

```diff
@@ -0,0 +1,255 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mackerelio/mackerel-client-go"
+)
+
+func resourceMackerelChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMackerelChannelCreate,
+		Read:   resourceMackerelChannelRead,
+		Update: resourceMackerelChannelUpdate,
+		Delete: resourceMackerelChannelDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"emails": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			// Field name may only contain lowercase alphanumeric characters & underscores.
+			"user_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"events": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
`url` cannot be used with `emails` and `user_ids`, should use `ConflictsWith`.

```suggestion
			"url": {
				Type:     schema.TypeString,
				Optional: true,
				ConflictsWith: []string{"emails", "user_ids"},
			},
```